### PR TITLE
Bugfixes for getting operative testnet nodes

### PR DESCRIPTION
**Pull request by cygnusv**
Based on #1175 

The systemd service file I'm using, in case it's useful:
```
[Unit]
Description="Run 'Ursula', a NuCypher Staking Node."

[Service]
Type=simple
Environment="NUCYPHER_KEYRING_PASSWORD=<your keyring password>"
Environment='NUCYPHER_WORKER_ETH_PASSWORD=<your account password>'
ExecStart=<virtualenv path>/bin/nucypher ursula run --teacher <teacher_uri>

[Install]
WantedBy=multi-user.target
```

### CODE REVIEW

File: nucypher/cli/actions.py [hunk 1]
```diff
@@ -296,6 +310,9 @@ def select_stake(stakeholder, emitter) -> Stake:
 
 def select_client_account(emitter, blockchain, prompt: str = None, default=0) -> str:
     enumerated_accounts = dict(enumerate(blockchain.client.accounts))
+    if len(enumerated_accounts) < 1:
+        emitter.echo("No ETH accounts were found.", fg='red', bold=True)
```
Comment (mswilkison) [inline nucypher/cli/actions.py]:
```suggestion
        emitter.echo("No ETH accounts were found.", color='red', bold=True)
```

File: nucypher/cli/actions.py [hunk 2]
```diff
@@ -304,6 +310,9 @@ def select_stake(stakeholder, emitter) -> Stake:
 
 def select_client_account(emitter, blockchain, prompt: str = None, default=0) -> str:
     enumerated_accounts = dict(enumerate(blockchain.client.accounts))
+    if len(enumerated_accounts) < 1:
```
Comment (derekpierre) [inline nucypher/cli/actions.py]: 👍

File: nucypher/cli/stake.py [hunk 3]
```diff
@@ -115,11 +115,10 @@ def stake(click_config,
...
 
+        fetch_registry = registry_filepath is None and not click_config.no_registry
         registry = None
-        fetch_registry = True
         if registry_filepath:
             registry = EthereumContractRegistry(registry_filepath=registry_filepath)
-            fetch_registry = False
         blockchain = BlockchainInterface(provider_uri=provider_uri, registry=registry, poa=poa)
 
         blockchain.connect(fetch_registry=False, sync_now=sync, emitter=emitter)
```
Comment (derekpierre) [inline nucypher/cli/stake.py]: We should use the `fetch_registry` value

```suggestion
        blockchain.connect(fetch_registry=fetch_registry, sync_now=sync, emitter=emitter)
```

File: nucypher/network/protocols.py [hunk 4]
```diff
@@ -32,6 +32,8 @@ def parse_node_uri(uri: str):
 
     if '@' in uri:
         checksum_address, uri = uri.split("@")
+        if checksum_address is None:
+            raise ValueError(f"{uri} is not a valid Teacher URI.")
```
Comment (derekpierre) [inline nucypher/network/protocols.py]:
```suggestion
            raise ValueError(f"{uri} is not a valid Teacher URI - no checksum address.")
```

File: nucypher/cli/stake.py (hunk 3 above)
Comment (cygnusv) [reply nucypher/cli/stake.py]: Ah yes, good catch! It's a problem of the rebase.

**Comment by cygnusv** [comment]
@michwill That bugfix is being currently developed by @KPrasch is his `service` branch. 

In any case, the problem you found starts when trying to read known nodes info. An ugly workaround is simply to remove the known nodes certificates and metadata.

**Pull request merged**